includes/Classifai/Providers/OpenAI/APIRequest.php

```diff
@@ -43,7 +52,26 @@ public function __construct( string $api_key = '' ) {
 	 */
 	public function get( string $url, array $options = [] ) {
 		$this->add_headers( $options );
-		return $this->get_result( wp_remote_get( $url, $options ) ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get
+
+		/**
+		 * Filter the response from OpenAI for a get request.
+		 *
+		 * @since x.x.x
+		 * @hook classifai_openai_api_response_get
+		 *
+		 * @param {string} $url Request URL.
+		 * @param {array} $options Request body options.
+		 * @param {string} $this->feature Feature name.
+		 *
+		 * @return {array} API response.
+		 */
+		return apply_filters(
```

Comment:
While I think it's fine to have a filter around the responses, I think even more valuable is having filters around the data we send into the request. In this case that would be the `$url` and `$options` parameters. I think having separate filters for each of those makes sense and will need to be done for all requests in this class.

Reply:
The `$options` are already coming through the filter of the `$body`.

Few [examples](https://github.com/10up/classifai/blob/cfc796455d0c4b682b770adf969d1854f6e227b6/includes/Classifai/Providers/OpenAI/ChatGPT.php),

* `classifai_chatgpt_excerpt_request_body`
* `classifai_chatgpt_title_request_body`
* `classifai_chatgpt_resize_content_request_body`

However, adding separate filters for `$options` still makes sense.